### PR TITLE
Unittests for aqua.evaluation/finetuning and handlers(evaluation/common/base)

### PR DESCRIPTION
**Pull request by mingkang111**
PR contains unittest for aqua.evaluation, aqua.finetuning, evaluation handler, common handler, and base handler.

# Coverage
Screenshot 2024-04-12 at 12 53 59

This PR also includes @lu-ohai 's contribution on `create` evaluation and FT.

### CODE REVIEW

File: ads/aqua/evaluation.py
Comment (mayoor) [inline ads/aqua/evaluation.py:1]: @mingkang111 , could you please reset this file to what is in main? I understand that it is only comments and formatting. But Would like only change to tests folder to the master before we make changes to any code of any kind.

Comment (mingkang111) [reply ads/aqua/evaluation.py:1]: Done.

**Comment by lu-ohai** [comment]
Just added the unit tests for create evaluation and fine tuning to this branch.

**Review by VipulMascarenhas — DISMISSED** [review]
lgtm 👍

**Review by mayoor — APPROVED** [review]

**Review by mrDzurb — DISMISSED** [review]

**Review by VipulMascarenhas — DISMISSED** [review]

**Review by VipulMascarenhas — DISMISSED** [review]

**Review by mrDzurb — DISMISSED** [review]

**Comment by mingkang111** [comment]
The failing operation test picking up `ray==2.3.0` which is incompatible with `pydantic==2.7.0`. Based on ray's repo https://github.com/ray-project/ray/issues/37019#issuecomment-1841864721 , `ray` is required to >=2.5 to be compatible with `pydantic>=2`. Therefore, we can remove the `pydantic` as test dependency for now and implement the checking by ourselves. Once ray can be installed with > 2.5, we go back to use `pydantic` again. 
cc: @ahosler

**Review by mayoor — APPROVED** [review]

**Review by VipulMascarenhas — APPROVED** [review]

**Pull request merged**